arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/floorfilter/FloorFilterView.kt

```diff
@@ -0,0 +1,1211 @@
+                selectedButtonBackgroundColor = getColor(R.styleable.FloorFilterView_selectedButtonBackgroundColor, DEFAULT_SELECTED_BUTTON_BACKGROUND_COLOR)
+                closeButtonBackgroundColor = getColor(R.styleable.FloorFilterView_closeButtonBackgroundColor, DEFAULT_CLOSE_BUTTON_BACKGROUND_COLOR)
+                searchBackgroundColor = getColor(R.styleable.FloorFilterView_searchBackgroundColor, DEFAULT_SEARCH_BACKGROUND_COLOR)
+
+                // size attributes
+                buttonHeightDp = getDimensionPixelSize(R.styleable.FloorFilterView_buttonHeight, 0).pixelsToDp(displayDensity)
+                buttonWidthDp = getDimensionPixelSize(R.styleable.FloorFilterView_buttonWidth, 0).pixelsToDp(displayDensity)
+                if (buttonHeightDp <= 0) {
+                    buttonHeightDp = DEFAULT_BUTTON_HEIGHT_DP
+                }
+                if (buttonWidthDp <= 0) {
+                    buttonWidthDp = DEFAULT_BUTTON_WIDTH_DP
+                }
+            } finally {
+                recycle()
+            }
+        }
+        init(context)
+    }
+
+    /**
+     * Initializes this FloorFilterView by inflating the layout and setting the [RecyclerView]
+     * adapter.
+     *
+     * @since 100.13.0
+     */
+    private fun init(context: Context) {
+        orientation = VERTICAL
+        inflate(context, R.layout.layout_floorfilterview, this)
+
+        floorsRecyclerView = findViewById(R.id.floorsRecyclerView)
+        floorListCloseButton = findViewById(R.id.floorListCloseButton)
+        siteFacilityButton = findViewById(R.id.siteFacilityButton)
+        siteFacilityButtonSeparator = findViewById(R.id.siteFacilityButtonSeparator)
+
+        addDataChangeListeners()
+        setupLevelsAdapter()
+        setupSiteFacilityButton()
+        setButtonPositions()
+        processUiParamUpdate()
+    }
+
+    /**
+     * Zooms to the selected [FloorFacility]. If no [FloorFacility] is selected, it will zoom to
+     * the selected [FloorSite].
+     *
+     * @since 100.13.0
+     */
+    fun zoomToSelection() {
+        floorFilterManager.zoomToSelection()
+    }
+
+    /**
+     * Returns the [maxDisplayLevels] used to determine the max amount of levels to show in the [floorsRecyclerView].
+     *
+     * The default is -1. Anything that is less than 1 will show all of the levels.
+     * Use [setMaxDisplayLevels] to change the max amount of levels to display.
+     *
+     * @since 100.13.0
+     */
+    fun getMaxDisplayLevels(): Int? {
+        return if (maxDisplayLevels < 0) {
+            null
+        } else {
+            maxDisplayLevels
+        }
+    }
+
+    /**
+     * The [Int] used to determine the max amount of levels to show in the list before scrolling.
+     *
+     * The default is -1. Anything that is less than 1 will show all of the levels.
+     * Use [getMaxDisplayLevels] to get the current value.
+     *
+     * @since 100.13.0
+     */
+    fun setMaxDisplayLevels(maxDisplayLevels: Int?) {
```

Comment:
Yeah, I think you are right. Calling scrollToSelectedLevel() is not necessary here. I will make maxDisplayLevels a public property.

Reply:
Please have a look at `setTextSize()` as well and see if we can also have it as a public property

Reply:
I would like to leave setTextSize() as a function so that the developer can pass in a unit and define the text size in SP instead of having to define it as pixels.